src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java

```diff
@@ -191,13 +191,18 @@ public Result createSession(HttpClient client, Command command)
                             return toReturn;
                         }).orElseThrow(() -> new SessionNotCreatedException(
                                 format("Unable to create new remote session. desired capabilities = %s", desired)));
-                    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                    } catch (NoSuchMethodException | IllegalAccessException e) {
                         throw new WebDriverException(format("It is impossible to create a new session "
                                         + "because 'createSession' which takes %s, %s and %s was not found "
                                         + "or it is not accessible",
                                 HttpClient.class.getSimpleName(),
                                 InputStream.class.getSimpleName(),
                                 long.class.getSimpleName()), e);
+                    } catch (InvocationTargetException e) {
+                        Throwable cause = e.getCause();
+                        throw new SessionNotCreatedException(
+                                format("Unable to create new remote session. desired capabilities = %s"
+                                        + "and reason is " + cause.getMessage(), desired));
```

Comment:
why not to use format specifiers for both arguments?

Reply:
Also, it might be handy to pass the original exception as `cause` constructor argument, so it would be easier to debug it.

Reply:
```
Throwable cause = e.getCause();
                        throw new SessionNotCreatedException(
                                format("Unable to create new remote session. desired capabilities = %s"
                                        + "and reason is %s", desired, cause != null ? cause.getMessage() : null,
                                        e.getTargetException()));
```

@mykola-mokhnach Do you mean it this way?

Reply:
almost. It's just instead of setting reason to null if cause is null you set it to the original exception instead

Reply:
also SessionNotCreatedException has the second constructor where one can provide the cause:

`SessionNotCreatedException(String msg, Throwable cause)`